www/js/app.js

```diff
@@ -29,13 +36,24 @@ class App extends React.Component {
   }
 }
 
+const local = new PouchDB('stop-here-db');
+const gateway = new Gateway(local);
+
+const storeBuilder = new StoreBuilder([ gateway.getMiddleware() ]);
+const store = storeBuilder.build();
+
 /* Requires cordova.js to already be loaded via <script> */
-const deviceReady = new Promise(resolve => {
-  document.addEventListener('deviceReady', resolve, false);
-});
+document.addEventListener('deviceReady', () => {
+
+  const network = new NetworkManager(store);
+  network.monitor();
 
-Promise.all([deviceReady, loadDb]).then( ([device, points]) => {
```

Comment:
I decided to change the `deviceReady` mechanism back to an event callback. Instead of waiting for all points to load, we render the UI first and then insert all the points using the `RELOAD` action